File: src/org/elixir_lang/sdk/erlang_dependent/SdkModificatorRootTypeConsumer.java
package org.elixir_lang.sdk.erlang_dependent;

import com.intellij.openapi.projectRoots.SdkModificator;
import com.intellij.openapi.roots.OrderRootType;
import com.intellij.openapi.vfs.VirtualFile;
import org.jetbrains.annotations.NotNull;

public  interface SdkModificatorRootTypeConsumer {
    void consume(@NotNull SdkModificator sdkModificator,
                 @NotNull VirtualFile[] configuredRoots,
                 @NotNull VirtualFile expandedInternalRoot,
                 @NotNull OrderRootType type);
}

File: src/org/elixir_lang/sdk/erlang_dependent/SdkModificatorRootTypeConsumer.kt
package org.elixir_lang.sdk.erlang_dependent

import com.intellij.openapi.projectRoots.SdkModificator
import com.intellij.openapi.roots.OrderRootType
import com.intellij.openapi.vfs.VirtualFile

interface SdkModificatorRootTypeConsumer {
    fun consume(
        sdkModificator: SdkModificator,
        configuredRoots: Array<VirtualFile?>?,
        expandedInternalRoot: VirtualFile,
        type: OrderRootType
    )
}
